awx/main/models/projects.py

```diff
@@ -542,9 +553,14 @@ def get_absolute_url(self, request=None):
     def get_ui_url(self):
         return urlparse.urljoin(settings.TOWER_URL_BASE, "/#/jobs/project/{}".format(self.pk))
 
+    def should_update_project(self):
+        if not self.project:
+            return False
+        return (self.job_type == 'check' or self.project.status == 'never updated')
```

Comment:
If `job_type` is not check, then this should further check that `self.project.current_update_id` is `self.pk`. This would be a much more iron-clad way to prevent a check type update conflicting with such an update.

This is because of tie-ins with some other active issues.